js/src/javascript/beautifier.js

```diff
@@ -664,6 +668,44 @@ function Beautifier(js_source_text, options) {
         output.space_before_token = opt.space_after_anon_function;
       }
 
+
+      // {
+      //   data() {}
+      // };
+      // vs
+      // {
+      //   data () {}
+      // }
+      if (current_token.text === '(') {
```

Comment:
There's a lot going on here.  Please add more tests to show more clearly what this code accomplishes, add comments to explain what the clauses cover, or both.

Reply:
The modified